### ISSUE TITLE
AndroidGraphicsLiveWallpaper deadlocks on resume

### ISSUE THREAD

**Issue by cypherdare** [libgdx/libgdx#5042]
#### Issue details
The application can deadlock  in a waiting condition when AndroidGraphicsLiveWallpaper resumes. I don't know the circumstances causing it. It used to be an extremely rare crash I've seen across all my live wallpapers the past several years. On my latest live wallpaper, it is affecting more than 1% of users. This is my first live wallpaper that uses the gyroscope or rotation vector sensor, so maybe it's triggering a race condition.

#### Reproduction steps/code
I haven't personally reproduced it. It affects over 1% of my users in my live wallpaper that uses the gyroscope/rotation vector sensor.

#### Version of LibGDX and/or relevant dependencies
1.9.7

#### Stacktrace
There are a few variations of this, depending on what we are resuming from. It seems like maybe `resume` is getting called on the UI thread, although that shouldn't happen. It might be an Android bug that we can work around by checking the current thread before waiting. 

```java
"main" prio=5 tid=1 Waiting
  | group="main" sCount=1 dsCount=0 obj=0x74a1baf0 self=0x7f86e96a00
  | sysTid=3100 nice=0 cgrp=bg_non_interactive sched=0/0 handle=0x7f8b285a98
  | state=S schedstat=( 7982502995 15063996268 37343 ) utm=516 stm=282 core=0 HZ=100
  | stack=0x7fd9f7b000-0x7fd9f7d000 stackSize=8MB
  | held mutexes=
  at java.lang.Object.wait! (Native method)
- waiting on <0x0f5a7e2b> (a java.lang.Object)
  at com.badlogic.gdx.backends.android.AndroidGraphicsLiveWallpaper.resume (AndroidGraphicsLiveWallpaper.java:139)
- locked <0x0f5a7e2b> (a java.lang.Object)
  at com.badlogic.gdx.backends.android.AndroidLiveWallpaper.onResume (AndroidLiveWallpaper.java:159)
  at com.badlogic.gdx.backends.android.AndroidLiveWallpaperService$AndroidWallpaperEngine.onResume (AndroidLiveWallpaperService.java:402)
  at com.badlogic.gdx.backends.android.AndroidLiveWallpaperService$AndroidWallpaperEngine.notifyVisibilityChanged (AndroidLiveWallpaperService.java:369)
  at com.badlogic.gdx.backends.android.AndroidLiveWallpaperService$AndroidWallpaperEngine.onVisibilityChanged (AndroidLiveWallpaperService.java:361)
  at android.service.wallpaper.WallpaperService$Engine.reportVisibility (WallpaperService.java:942)
  at android.service.wallpaper.WallpaperService$Engine.doVisibilityChanged (WallpaperService.java:913)
  at android.service.wallpaper.WallpaperService$IWallpaperEngineWrapper.executeMessage (WallpaperService.java:1238)
  at com.android.internal.os.HandlerCaller$MyHandler.handleMessage (HandlerCaller.java:37)
  at android.os.Handler.dispatchMessage (Handler.java:102)
  at android.os.Looper.loop (Looper.java:159)
  at android.app.ActivityThread.main (ActivityThread.java:6146)
  at java.lang.reflect.Method.invoke! (Native method)
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:886)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:776)
```

**Comment by Zomby2D** [comment]
I believe PR #4267 was aiming at solving this particular issue.

**Comment by cypherdare** [comment]
I haven't encountered this with a live wallpaper in a long time, and it may have been fixed with #5042.
